domenetjenester/iay/src/main/java/no/nav/foreldrepenger/abakus/domene/iay/InntektArbeidYtelseRepository.java

```diff
@@ -299,22 +276,60 @@ public GrunnlagReferanse lagre(KoblingReferanse koblingReferanse, Arbeidsforhold
         Objects.requireNonNull(inntektsmeldingerList, "inntektsmelding"); // NOSONAR
         InntektArbeidYtelseGrunnlagBuilder builder = opprettGrunnlagBuilderFor(koblingReferanse);
 
-        oppdaterBuilderMedNyeInntektsmeldinger(informasjonBuilder, inntektsmeldingerList, builder);
+        var utdaterteInntektsmeldingerJournalposter = oppdaterBuilderMedNyeInntektsmeldinger(informasjonBuilder, inntektsmeldingerList, builder);
+
+        var utdaterteInntektsmeldinger = inntektsmeldingerList.stream().filter(it -> utdaterteInntektsmeldingerJournalposter.contains(it.getJournalpostId())).collect(Collectors.toList());
 
         InntektArbeidYtelseGrunnlag build = builder.build();
+        var utdatertBuilder = InntektArbeidYtelseGrunnlagBuilder.oppdatere(build);
+        lagreDeaktivertGrunnlagMedUtdaterteInntektsmeldinger(koblingReferanse, informasjonBuilder, utdaterteInntektsmeldinger, utdatertBuilder);
         lagreOgFlush(koblingReferanse, build);
+
         return build.getGrunnlagReferanse();
     }
 
-    /** Legger til nye inntektsmeldinger på angitt IAYG builder. */
-    public void oppdaterBuilderMedNyeInntektsmeldinger(ArbeidsforholdInformasjonBuilder informasjonBuilder,
-                                                       List<Inntektsmelding> nyeInntektsmeldinger,
-                                                       InntektArbeidYtelseGrunnlagBuilder targetBuilder) {
+    public void lagreDeaktivertGrunnlagMedUtdaterteInntektsmeldinger(KoblingReferanse koblingReferanse, ArbeidsforholdInformasjonBuilder informasjonBuilder,
+                                                                      List<Inntektsmelding> utdaterteInntektsmeldinger,
+                                                                      InntektArbeidYtelseGrunnlagBuilder utdatertBuilder) {
+        utdatertBuilder.medDeaktivert();
 
-        if(nyeInntektsmeldinger.isEmpty()) {
+        if (utdaterteInntektsmeldinger.isEmpty()) {
             return; // quick exit, ingenting nytt å gjøre her
         }
 
+        final InntektsmeldingAggregat inntektsmeldinger = utdatertBuilder.getInntektsmeldinger();
+        for (Inntektsmelding inntektsmelding : utdaterteInntektsmeldinger) {
+            // Kommet inn inntektsmelding på arbeidsforhold som vi har gått videre med uten inntektsmelding?
+            if (informasjonBuilder.kommetInntektsmeldingPåArbeidsforholdHvorViTidligereBehandletUtenInntektsmelding(inntektsmelding)) {
```

Comment:
Er det nødvendig å gjere denne sjekken for dei utdaterte inntektsmeldingene også? Desse vil vel per def ha blitt erstatta av ei inntektsmelding og då vil vel dette ha blitt gjort for denne?

Reply:
Fjerner dette